lib/dhl_uk/errors/soap_response_error.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module DhlUk
+  module Errors
+    class SoapResponseError < Error
+      attr_reader :payload
+
+      def initialize(response, payload)
+        @payload = payload
+  # TODO: parse error messages
```

Comment:
Parse or not to parse... 🙂

Reply:
I've changed my mind completely on this having looked at the way the main app handles DHLExpress errors. 
Just passing back the response with no comment and raising error when request failed entirely.